feat(tabs): Add count and small variant

tihuan opened the pull request:
## Summary

**Structural Element (Gene)**
Shortcut ticket: [161558](https://app.shortcut.com/sci-design-system/story/161558/complete-tabs-component)
Copy ticket description here:

**Add Small variant to Tabs component:**
1. Label font should be `fontBodyXs`
2. Label font weight should be `fontWeightSemibold`

**Prop to remove persistent underline:**
1. Add prop option to enable or disable persistent underline from Tabs component (both Large and Small variants)

**Add counter to Large variant:**
1. Add prop option to add a Counter to Tabs component
2. For `Large` variant, counter should be `fontBodyXs`
3. For `Large` variant, there should be `spaceL` between Label and Counter
4. Counter should be color `gray500`
5. Counter should be center aligned to Label

**Add counter to Small variant:**
1. Add prop option to add a Counter to Tabs component
2. For `Small` variant, counter should be `fontBodyXxs`
3.  For `Large` variant, there should be `spaceM` between Label and Counter
4. Counter should be color `gray500`
5. Counter should be center aligned to Label

<img width="1021" alt="Screen Shot 2022-02-25 at 12 08 07 PM" src="https://user-images.githubusercontent.com/6309723/155804248-a9e1d4c0-2c96-411b-8369-00ffa5733b66.png">

## Checklist

- [x] Default Story in Storybook
- [x] LivePreview Story in Storybook
- [x] Test Story in Storybook
- [x] Tests written
- [x] Variables from `defaultTheme.ts` used wherever possible
- [x] If updating an existing component, depreciate flag has been used where necessary
- [ ] Chromatic build verified by @chanzuckerberg/sds-design

mdjmoore reviewed (approved):
┗(＾0＾)┓

Can you add the Tab v5 upgrade to [this ticket](https://app.shortcut.com/sci-design-system/story/185930/material-ui-mui-v5-todos) and [this issue](https://github.com/chanzuckerberg/sci-components/issues/120)

TIA

tihuan commented:
@clarsen-czi Something for Monday:

Thanks so much for the great feedback! Updated code accordingly 👌 

The only one that's gonna be a huge lift to implement is adding gray300 underline on hovering inactive tab (essentially implement our own tab indicator from scratch). So I added on hover highlight with gray600 as a compromise. What do you think?

Thank you and have an amazing weekend!

<img width="1197" alt="Screen Shot 2022-03-04 at 5 31 02 PM" src="https://user-images.githubusercontent.com/6309723/156862176-7b0d4a9b-9bc3-4095-bfd4-aa792de18e9c.png">

![demo](https://user-images.githubusercontent.com/6309723/156862179-ac542db6-242e-49cc-8b65-f6b839051127.gif)

tihuan commented:
> ┗(＾0＾)┓
> 
> Can you add the Tab v5 upgrade to [this ticket](https://app.shortcut.com/sci-design-system/story/185930/material-ui-mui-v5-todos) and [this issue](https://github.com/chanzuckerberg/sci-components/issues/120)
> 
> TIA

Oh that's right! Just updated 🎉 Thank you for the reminder!

tihuan commented:
> @clarsen-czi Something for Monday:
> 
> Thanks so much for the great feedback! Updated code accordingly 👌
> 
> The only one that's gonna be a huge lift to implement is adding gray300 underline on hovering inactive tab (essentially implement our own tab indicator from scratch). So I added on hover highlight with gray600 as a compromise. What do you think?
> 
> Thank you and have an amazing weekend!
> 
> <img alt="Screen Shot 2022-03-04 at 5 31 02 PM" width="1197" src="https://user-images.githubusercontent.com/6309723/156862176-7b0d4a9b-9bc3-4095-bfd4-aa792de18e9c.png">
> 
> ![demo](https://user-images.githubusercontent.com/6309723/156862179-ac542db6-242e-49cc-8b65-f6b839051127.gif)

Hi @clarsen-czi gentle ping thank you 😄 🙏

clarsen-czi commented:
> 

Sorry for the delay @tihuan ! Let me bring this back to the design team and see how we want to move forward.

tihuan commented:
Thanks so much, Connor! Yeah sorry that it turned out to take a lot more effort to do that 😵

tihuan commented:
Just changed to black! PTAL thank you 🚀 

![demo](https://user-images.githubusercontent.com/6309723/157512857-0d9fea4e-a72a-4839-b388-a4fd8a2b5357.gif)

tihuan commented:
@clarsen-czi PTAL thank you 😄 !

clarsen-czi reviewed (approved):
LGTM! Thanks Timmy!

tihuan commented:
Yayy! Thanks again, Connor and Meagan 🎉 Merging nowww
